### PR TITLE
python311Packages.amazon-ion: 0.10.0 -> 0.11.2

### PR DESCRIPTION
**Pull request by fabaff**
Diff: https://github.com/amzn/ion-python/compare/refs/tags/v0.10.0...v0.11.2

Changelog: https://github.com/amazon-ion/ion-python/releases/tag/v0.11.2

## Description of changes
Fix build (https://hydra.nixos.org/build/241207251)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 270380` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>python310Packages.ionhash</li>
    <li>python310Packages.ionhash.dist</li>
    <li>python310Packages.pyqldb</li>
    <li>python310Packages.pyqldb.dist</li>
    <li>python311Packages.ionhash</li>
    <li>python311Packages.ionhash.dist</li>
    <li>python311Packages.pyqldb</li>
    <li>python311Packages.pyqldb.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.amazon-ion</li>
    <li>python310Packages.amazon-ion.dist</li>
    <li>python311Packages.amazon-ion</li>
    <li>python311Packages.amazon-ion.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/amazon-ion/default.nix [hunk 1]
```diff
@@ -1,54 +1,69 @@
...
-  version = "0.10.0";
-  format = "setuptools";
+  version = "0.11.2";
+  pyproject = true;
 
   disabled = pythonOlder "3.7";
 
-  # test vectors require git submodule
   src = fetchFromGitHub {
     owner = "amzn";
```
Comment (anthonyroussel) [inline pkgs/development/python-modules/amazon-ion/default.nix:21]:
```suggestion
    owner = "amazon-ion";
```

Repository has been moved to https://github.com/amazon-ion/ion-python

**Review by anthonyroussel — COMMENTED** [review]
Result of `nixpkgs-review pr 270380` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>4 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python310Packages.ionhash: <a href="https://termbin.com/8xcn">plain log</a></li>
<li>python310Packages.pyqldb: log was empty</li>
<li>python311Packages.ionhash: <a href="https://termbin.com/2p8b">plain log</a></li>
<li>python311Packages.pyqldb: log was empty</li>
  </ul>
</details>

<details>
<summary>
4 packages built:
</summary>
<ul>
<li>python310Packages.amazon-ion</li>
<li>python310Packages.amazon-ion.dist</li>
<li>python311Packages.amazon-ion</li>
<li>python311Packages.amazon-ion.dist</li>
</ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 270380` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>python310Packages.ionhash</li>
    <li>python310Packages.ionhash.dist</li>
    <li>python310Packages.pyqldb</li>
    <li>python310Packages.pyqldb.dist</li>
    <li>python311Packages.ionhash</li>
    <li>python311Packages.ionhash.dist</li>
    <li>python311Packages.pyqldb</li>
    <li>python311Packages.pyqldb.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.amazon-ion</li>
    <li>python310Packages.amazon-ion.dist</li>
    <li>python311Packages.amazon-ion</li>
    <li>python311Packages.amazon-ion.dist</li>
  </ul>
</details>

**Pull request merged**